### PR TITLE
Added instance of account object to session class

### PR DESCRIPTION
**Pull request by dafuga**

### CODE REVIEW

File: src/session.ts [hunk 1]
```diff
@@ -17,10 +17,12 @@ import {
     ResolvedTransaction,
     SigningRequest,
 } from 'eosio-signing-request'
+import { Account } from '@wharfkit/account'
+
 import zlib from 'pako'
 
-import {ChainDefinition, ChainDefinitionType, Fetch} from './types'
-import {getFetch} from './utils'
+import { ChainDefinition, ChainDefinitionType, Fetch } from './types'
```
Comment (dafuga) [inline src/session.ts:24]: Not sure why our IDEs are formatting code differently.. I am using CS Code now 😅

Comment (ericpassmore) [reply src/session.ts:24]: Regarding brackets: you can try adding this to the end of the eslint file. Current set to warn. 
```
prettierOptions: {
    bracketSpacing: false,
},
```

Comment (aaroncox) [reply src/session.ts:24]: Using `make format` seems to follow the rules. One thing to check with VSCode is that you have the Prettier extension installed:

image

I'm not sure whether or not it obeys those rules by default. We do have a `.prettierrc` file in the root that your editor should follow:

https://github.com/wharfkit/session/blob/83b598de7f1512f758f8b5cb768de9d49846bc1b/.prettierrc#L1-L8

**Comment by aaroncox** [comment]
I switched this PR to merge into the `v0.2.x` branch, which has some changes. This caused a few conflicts. 

One notable one is that I renamed the `account` getter on the Session to `accountName`, in anticipation of you using the `account` getter to return the account.

**Comment by dafuga** [comment]
@aaroncox done!

File: test/tests/session.ts [hunk 2]
```diff
@@ -307,5 +308,13 @@ suite('session', function () {
...
         assert.isTrue(session.actor.equals(expectedPermission.actor))
         assert.isTrue(session.permission.equals(expectedPermission.permission))
+        assert.instanceOf(
+            session.account,
+            Account
+        ) 
+        assert.equal(
+            session.permission,
+            PermissionLevel.from(mockSessionOptions.permissionLevel!).permission
+        )
```
Comment (aaroncox) [inline test/tests/session.ts:318]: This permission test is a duplicate of what's above the `instanceOf` check.

File: yarn.lock [hunk 3]
```diff
@@ -600,6 +618,21 @@ ajv@^6.10.0, ajv@^6.12.4:
     json-schema-traverse "^0.4.1"
     uri-js "^4.2.2"
 
+anchor-link@^3.5.1:
```
Comment (aaroncox) [inline yarn.lock:621]: Hrm, why does the Account Kit need anchor-link?

Might be worth cleaning up the Account Kit dependencies before merging it in here to the Session Kit, so we don't have much in the way of extra dependencies.

Ideally - the only thing the Account Kit should add as a dependency is itself.

**Comment by aaroncox** [comment]
I'm also swapping this to the `0.3.x` branch now, since this would be a breaking change to `0.2.x`.

**Comment by aaroncox** [comment]
This line is also going to break the session kit 🤔 

https://github.com/wharfkit/account/blob/mvp/package.json#L20

Do we need that private prototype as part of the account kit?

**Comment by aaroncox** [comment]
Just as a thought about this PR - I think at this very moment we should leave this specific PR unmerged until the Account Kit itself is functional within the Session Kit. The reason being is that the Session Kit is rapidly approaching a point where it'll actually be usable in development, and if we include an Account Kit instance that's unusable - that may create a weird situation for developers who stumble upon it. 

The reason I'd consider it unusable at this point is because of the MockSession it's using for development purposes, and as I understand it, the MockSession in the temporary Contract Kit component will be replaced with a real session once the Contract Kit is available and implemented?

**Comment by dafuga** [comment]
Yeah that's a good point, most of what the Account kit does will be pretty useless until the contract kit is ready.

**Pull request closed without merging**